src/main/java/dev/warrant/model/object/ListResult.java

```diff
@@ -0,0 +1,45 @@
+package dev.warrant.model.object;
+
+public class ListResult<T extends WarrantObject> {
```

Comment:
Do we need to limit `T` to only those types that implement `WarrantObject`? If we remove this restriction, we can also use `ListResult` for the query results: `ListResult<QueryResult>`

Reply:
I think this can be any type so I updated it to just `T` but I didn't update the query results return type for the reason I listed below